File: app/src/main/java/seveida/firetvforreddit/response/objects/Source.java
package seveida.firetvforreddit.response.objects;

import com.squareup.moshi.Json;

public class Source {

    @Json(name = "url")
    public String url;
    @Json(name = "width")
    public int width;
    @Json(name = "height")
    public int height;

    /**
     * No args constructor for use in serialization
     *
     */
    public Source() {
    }

    /**
     *
     * @param height
     * @param width
     * @param url
     */
    public Source(String url, int width, int height) {
        super();
        this.url = url;
        this.width = width;
        this.height = height;
    }

}

File: app/src/main/java/seveida/firetvforreddit/response/objects/Source.kt
package seveida.firetvforreddit.response.objects

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Source(
    @Json(name = "url") val url: String,
    @Json(name = "width") val width: Int = 0,
    @Json(name = "height") val height: Int = 0
)
